src/main/DynamicConfig.ts

```diff
@@ -299,13 +299,46 @@ export class DynamicConfig implements IDynamicConfig {
     }
 
     public async getRemoteValue<T>(key: string, type?: ObjectType): Promise<T> {
-        return this.source(key).then((source: ISource) => {
-            if (source.key !== undefined) {
-                return this.getValueFromResolver<T>(source.key, 'remote', type)
-            } else {
-                throw new errors.ResolverUnavailable(key)
-            }
-        })
+        const source = await this.source(key) // get source for key
```

Comment:
For readability it could be nice to add type annotations to `source`, `resolvedConfig`,  `value` , `builtValue`